cookbooks/solr/recipes/default.rb

```diff
@@ -0,0 +1,140 @@
+#
+# Cookbook Name:: solr
+# Recipe:: default
+#
+# We specify what version we want below.
+use_default_java = false
+java_package_name = "dev-java/icedtea-bin"
+java_version = node['solr']['java_version']
+java_eselect_version = node['solr']['java_eselect_version']
+solr_version = node['solr']['solr_version']
+solr_file = "solr-#{solr_version}.tgz"
+solr_url = "https://archive.apache.org/dist/lucene/solr/#{solr_version}/#{solr_file}"
+core_name = node['solr']['core_name']
+
+username = node['dna']['users'].first['username']
+
+# Install Solr
+if ('solo' == node['dna']['instance_role'])  ||
+  ( ('util' == node['dna']['instance_role']) && ('solr' == node['dna']['name']) )
+
+  unless use_default_java
+    Chef::Log.info "Updating Java JDK"
+    enable_package java_package_name do
+      version java_version
+      unmask true
+    end
+
+    package java_package_name do
+      version java_version
+      action :upgrade
+    end
+
+    execute "Set the default Java version to #{java_eselect_version}" do
+      command "eselect java-vm set system #{java_eselect_version}"
+      action :run
+    end
+  end
+
+  directory "/var/run/solr" do
+    action :create
+    owner username
+    group username
+    mode 0755
+  end
+
+  directory "/var/log/engineyard/solr" do
+    action :create
+    owner username
+    group username
+    mode 0755
+    recursive true
+  end
+
+  template "/engineyard/bin/solr" do
+    source "solr.erb"
+    owner username
+    group username
+    mode 0755
+    variables({
+      :rails_env => node['dna']['environment']['framework_env']
+    })
+  end
+
+  template "/etc/monit.d/solr.monitrc" do
+    source "solr.monitrc.erb"
+    owner username
+    group username
+    mode 0644
+  end
+
+  remote_file "/data/#{solr_file}" do
+    source solr_url
+    owner username
+    group username
+    mode 0644
+    backup 0
+    not_if { FileTest.exists?("/data/#{solr_file}") }
+  end
+
+  execute "unarchive solr-to-install" do
```

Comment:
We can probably merged the two into a single `execute[]` statement with `-C <dir>` and `--strip-components=1`